manifests/basic_ucp/ucp-armada.yaml

```diff
@@ -0,0 +1,505 @@
+
+          [app:admin_service]
+          use = egg:keystone#admin_service
+
+          [pipeline:public_api]
+          # The last item in this pipeline must be public_service or an equivalent
+          # application. It cannot be a filter.
+          pipeline = cors sizelimit url_normalize request_id admin_token_auth build_auth_context token_auth json_body ec2_extension public_service
+
+          [pipeline:admin_api]
+          # The last item in this pipeline must be admin_service or an equivalent
+          # application. It cannot be a filter.
+          pipeline = cors sizelimit url_normalize request_id admin_token_auth build_auth_context token_auth json_body ec2_extension s3_extension admin_service
+
+          [pipeline:api_v3]
+          # The last item in this pipeline must be service_v3 or an equivalent
+          # application. It cannot be a filter.
+          pipeline = cors sizelimit url_normalize request_id admin_token_auth build_auth_context token_auth json_body ec2_extension_v3 s3_extension service_v3
+
+          [app:public_version_service]
+          use = egg:keystone#public_version_service
+
+          [app:admin_version_service]
+          use = egg:keystone#admin_version_service
+
+          [pipeline:public_version_api]
+          pipeline = cors sizelimit url_normalize public_version_service
+
+          [pipeline:admin_version_api]
+          pipeline = cors sizelimit url_normalize admin_version_service
+
+          [composite:main]
+          use = egg:Paste#urlmap
+          /v2.0 = public_api
+          /v3 = api_v3
+          / = public_version_api
+
+          [composite:admin]
+          use = egg:Paste#urlmap
+          /v2.0 = admin_api
+          /v3 = api_v3
+          / = admin_version_api
+    replicas: 2
+    images:
+      db_sync: kolla/ubuntu-source-keystone:2.0.2
+      api: kolla/ubuntu-source-keystone:2.0.2
+    labels:
+      node_selector_key: ucp-control-plane
+      node_selector_value: enabled
+  source:
+    type: git
+    location: git://git.openstack.org/openstack/openstack-helm
+    subpath: keystone
+  dependencies:
+    - helm-toolkit
+---
+schema: armada/Chart/v1
+metadata:
+  schema: metadata/Document/v1
+  name: maas-postgresql
+data:
+  chart_name: maas-postgresql
+  release: maas-postgresql
+  namespace: maas
+  install:
+    no_hooks: false
+  upgrade:
+    no_hooks: false
+  pre:
+    delete: []
+    create: []
+  post:
+    delete: []
+    create: []
+  values:
+    development:
+      enabled: false
+    labels:
+      node_selector_key: ucp-control-plane
+      node_selector_value: enabled
+  source:
+    type: git
+    location: git://git.openstack.org/openstack/openstack-helm-addons
+    subpath: postgresql
+  dependencies: []
+---
+schema: armada/Chart/v1
+metadata:
+  schema: metadata/Document/v1
+  name: maas
+data:
+  chart_name: maas
+  release: maas
+  namespace: maas
+  install:
+    no_hooks: false
+  upgrade:
+    no_hooks: false
+  pre:
```

Comment:
you dont need to declare these pre keywor if they are not been used. also if the pre is not under the upgrade object it will not register the actions.